pkgs/servers/pulseaudio/default.nix

```diff
@@ -114,7 +114,10 @@ stdenv.mkDerivation rec {
     ];
 
   postInstall = lib.optionalString libOnly ''
-    rm -rf $out/{bin,share,etc,lib/{pulse-*,systemd}}
+    find $out/share -maxdepth 1 -mindepth 1 ! -name "vala" -prune -exec rm -r {} \;
+    find $out/share/vala -maxdepth 1 -mindepth 1 ! -name "vapi" -prune -exec rm -r {} \;
+    [ -d $out/lib/systemd ] && rm -r $out/lib/systemd
```

Comment:
```suggestion
    ${lib.optionalString useSystemd "rm -r $out/lib/systemd"}
```

Reply:
Just tested this, and it seems useSystemd is true for libpulseaudio, but this directory is not created, so the build fails.

Reply:
Yea, this is a bit more complicated - `libpulseaudio` is built with `useSystemd = true` (by default) and `libOnly = true`. Because `(!libonly)` is false, it makes `systemd` not get added to the `buildInputs`, see:


https://github.com/NixOS/nixpkgs/blob/6ef4f522d63f22b40004319778761040d3197390/pkgs/servers/pulseaudio/default.nix#L52-L56

And in turn that `systemd` directory is not created. If so, perhaps you can remove this line altogether, and we can count on:

https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/blob/6a487c5244563f49a025088de3510c80824fc99e/src/daemon/meson.build#L144-159

And our `buildInputs` conditional, that this directory won't exist in this part of `postInstall`.